### DISCUSSION TITLE
How to set UV coordinates per triangle

### DISCUSSION THREAD

**Discussion by royvelich** [pygfx/pygfx#568] [Q&A]
Hi,
Let's say that I have a simple 3D cube, so each face is made up of two triangles. How can I set separate UV coordinates per triangle? Is it possible to do it without duplicating vertices?

**Comment by almarklein** [comment]
> Is it possible to do it without duplicating vertices?

No. Our builtin box geometry consists of 6 faces (2 triangles each), and each face has its own set of vertices exactly for this reason. From your question I understand that you want per-triangle texture coords, so in that case each triangle would need its own vertices.

**Comment by Korijn** [comment]
To be clear this is a general limitation in graphics software, it's not specific to pygfx or wgpu.

**Comment by royvelich** [comment]
@Korijn @almarklein 
Thanks guys
